pandas/core/indexing.py

```diff
@@ -731,11 +731,15 @@ def _getitem_lowerdim(self, tup: Tuple):
             raise IndexingError("Too many indexers. handle elsewhere")
 
         for i, key in enumerate(tup):
-            if is_label_like(key) or isinstance(key, tuple):
+            if is_label_like(key):
```

Comment:
can you add an assert

Reply:
im ambivalent about this, since optimization is a big part of the goal here